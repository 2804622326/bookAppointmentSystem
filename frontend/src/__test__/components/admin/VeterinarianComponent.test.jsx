import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import VeterinarianComponent from '../../../components/admin/VeterinarianComponent.jsx';

jest.mock('../../../components/veterinarian/VeterinarianService.js', () => ({
  getVeterinarians: jest.fn(() => Promise.resolve({ data: [] })),
}));

jest.mock('../../../components/common/Paginator.jsx', () => () => <div data-testid="paginator" />);
jest.mock('../../../components/user/UserFilter.jsx', () => () => <div data-testid="user-filter" />);
jest.mock('../../../components/modals/DeleteConfirmationModal.jsx', () => () => <div data-testid="delete-modal" />);
jest.mock('../../../components/veterinarian/VetEditableRows.jsx', () => () => <tr data-testid="edit-row" />);

test('shows no data message when there are no veterinarians', async () => {
  const { asFragment } = render(<VeterinarianComponent />);
  await waitFor(() => screen.getByText(/no.*veterinarian data.*available at the moment/i));
  expect(screen.getByText(/no.*veterinarian data.*available at the moment/i)).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
