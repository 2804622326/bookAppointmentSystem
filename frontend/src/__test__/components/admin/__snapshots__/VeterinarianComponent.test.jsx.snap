// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows no data message when there are no veterinarians 1`] = `
<DocumentFragment>
  <main
    class="mt-2"
  >
    <div
      class="text-center mt-5"
    >
      <h5>
        No  veterinarian data  available at the moment
      </h5>
    </div>
  </main>
</DocumentFragment>
`;
